shell-integration/pwsh/cmdex.ps1

```
@@ -0,0 +1,77 @@
+# everything launched from here on is protected.
+#
+# This does NOT protect against code that runs IN this session rather than
+# as a separate process — a dot-sourced profile script, another function,
+# Invoke-Expression — since a global variable has no privacy from other code
+# sharing the same session; there's no fix for that at this layer, and it
+# isn't a materially bigger hole regardless (such code could already do far
+# worse than spoof a copy-output marker).
+if (-not (Test-Path Variable:\global:cmdexNonce)) {
+    $cmdexNonceFile = $env:CMDEX_OSC_NONCE_FILE
+    if ($cmdexNonceFile -and (Test-Path $cmdexNonceFile)) {
+        $global:cmdexNonce = Get-Content -Path $cmdexNonceFile -Raw
```

PSScriptAnalyzer (1.25.0)

[warning] Found global variable 'global:cmdexNonce'.

(PSAvoidGlobalVars)